pkg/cmd/readyz/readyz.go

```diff
@@ -0,0 +1,187 @@
+package readyz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.etcd.io/etcd/client/pkg/v3/transport"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc"
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultEndpoint        = "https://localhost:2379"
+	defaultHTTPDialTimeout = 2 * time.Second
+	// keepalive defaults used by kube-apiserver
+	// https://github.com/kubernetes/apiserver/blob/de6ba2aa0a752d077719fcee186291d7af73a825/pkg/storage/storagebackend/factory/etcd3.go#L55-L56
+	keepaliveTime     = 30 * time.Second
+	keepaliveTimeout  = 10 * time.Second
+	DefaultListenPort = 7443
+)
+
+type readyzOpts struct {
```

Comment:
missing serving cert/key

Reply:
I was reusing the on disk etcd client cert/key files for the readyz server certs/key to keep the changes to a min:
```Go
err = server.ServeTLS(ln, r.clientCertFile, r.clientKeyFile)
``` 
That works fine but I'm guessing I should probably just issue a separate self-signed cert/key pair here, e.g:
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/cmd/render/certs.go#L21-L23

Just need to re-format it for how the TLS config expects it.

Reply:
client cert are not the same as serving certs.  They have different usages signed by the CA